core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/CaseSensitivitySpec.scala

```diff
@@ -81,56 +97,62 @@ class CaseSensitivitySpec extends AnyFreeSpec with Matchers with TestDriverCompo
 
         "should return all types in uncached metadata requests" in withTestDataServices { _ =>
           // save exemplar data
-          runAndWait(entityQuery.save(testWorkspace.workspace, exemplarData))
-          // get provider
-          val provider = new LocalEntityProvider(EntityRequestArguments(testWorkspace.workspace, testContext),
-                                                 slickDataSource,
-                                                 false,
-                                                 testConf.getDuration("entities.queryTimeout"),
-                                                 "metricsBaseName"
+          runAndWait(
+            compactEntityQuery.batchWriteEntities(testWorkspace.workspace.workspaceIdAsUUID,
+                                                  exemplarData,
+                                                  insertOnly = false
+            )
           )
+          // get provider
+          val provider = defaultProvider
+
           // get metadata
-          val metadata = provider.entityTypeMetadata(false, testContext).futureValue
+          val metadata = provider.entityTypeMetadata(useCache = false, testContext).futureValue
           metadata.keySet shouldBe exemplarTypes
         }
 
         "should return all types in cached metadata requests" in withTestDataServices { services =>
           // save exemplar data
-          runAndWait(entityQuery.save(testWorkspace.workspace, exemplarData))
+          runAndWait(
+            compactEntityQuery.batchWriteEntities(testWorkspace.workspace.workspaceIdAsUUID,
+                                                  exemplarData,
+                                                  insertOnly = false
+            )
+          )
 
           // assert cache is not yet populated
-          runAndWait(entityCacheQuery.entityCacheStaleness(testWorkspace.workspace.workspaceIdAsUUID)) shouldBe empty
+          runAndWait(compactEntityQuery.getCachedKeys(testWorkspace.workspace.workspaceIdAsUUID)) shouldBe empty
 
           // get metadata
           val metadata =
             services.entityService.entityTypeMetadata(testWorkspace.wsName, useCache = true).futureValue
           metadata.keySet shouldBe exemplarTypes
 
-          // assert cache is populated and up-to-date
-          runAndWait(entityCacheQuery.entityCacheStaleness(testWorkspace.workspace.workspaceIdAsUUID)) should contain(0)
+          // assert cache is populated
+          val cachedKeys = runAndWait(compactEntityQuery.getCachedKeys(testWorkspace.workspace.workspaceIdAsUUID))
+          cachedKeys should not be empty
 
           // get types from cache and verify
-          val cachedTypes = runAndWait(entityTypeStatisticsQuery.getAll(testWorkspace.workspace.workspaceIdAsUUID))
-          cachedTypes.keySet shouldBe exemplarTypes
+          val cachedTypes = cachedKeys.map(_.entityType)
+          cachedTypes should contain theSameElementsAs exemplarTypes
```

Comment:
a subtle assertion change here from `shouldBe` to `should contain theSameElementsAs` because we're now asserting on a Seq instead of a Set